packages/web-lib/contexts/useWeb3.tsx

```diff
@@ -121,8 +122,8 @@ export const Web3ContextApp = ({
 					})
 					.catch((): void => set_hasDisableAutoChainChange(true));
 			} else {
-				if (newChainID in chains) {
-					const chainSwap = chains[newChainID]?.chain_swap;
+				if (newChainID in chains.getAll()) {
+					const chainSwap = chains.get(newChainID)?.chain_swap;
```

Comment:
In https://github.com/yearn/web-lib/pull/164/files#r1065445099 I'm referring to this here, if the hook itself would take a `chainID` then we couldn't do this. Therefore, the hook just returns a few functions that can be used to get the chain details